File: back/src/main/java/fr/uha/ensilan/concours/back/dto/game/GameDto.java
package fr.uha.ensilan.concours.back.dto.game;

import fr.uha.ensilan.concours.back.domain.game.GameState;
import fr.uha.ensilan.concours.back.dto.question.QuestionDto;
import fr.uha.ensilan.concours.back.dto.question.QuestionSetDto;
import fr.uha.ensilan.concours.back.dto.user.UserDto;
import lombok.Data;

import java.util.Set;
import java.util.UUID;

@Data
public class GameDto {
    private UUID id;
    private String name;
    private GameState status;
    private QuestionSetDto questionSet;
    private QuestionDto currentQuestion;
    private Set<UserDto> players;
    private Set<UserDto> waitingPlayers;
}

File: front/src/app/core/domain/games/game.dto.ts
import {GameStateEnum} from "@domain/games/game-state.enum";
import {QuestionSetDto} from "@domain/questions/question-set.dto";
import {QuestionDto} from "@domain/questions/question.dto";
import {UserDto} from "@domain/user.dto";

export class GameDto {
  id: string;
  name: string;
  status: GameStateEnum;
  questionSet: QuestionSetDto;
  currentQuestion: QuestionDto;
  players: UserDto[];
  waitingPlayers: UserDto[];
}
